core/src/test/java/org/elasticsearch/search/sort/GeoDistanceSortBuilderTests.java

```diff
@@ -79,13 +81,13 @@ public static GeoDistanceSortBuilder randomGeoDistanceSortBuilder() {
             result.geoDistance(geoDistance(result.geoDistance()));
         }
         if (randomBoolean()) {
-            result.unit(unit(result.unit()));
+            result.unit(ESTestCase.randomValueOtherThan(result.unit(), () -> randomFrom(DistanceUnit.values())));
```

Comment:
This is similar to my comments in https://github.com/elastic/elasticsearch/pull/18042, I don't really mind but I think it would be nicer to just set a random value here, instead of using this "except one" helper method here. Just makes it easier to read.